### PR TITLE
V3: type hints for operations

### PR DESCRIPTION
**Pull request by stone-w4tch3r**
Type hints for all operations, V3 version

### CODE REVIEW

**Comment by stone-w4tch3r** [comment]
Looks like some check are broken :(

**Comment by Fizzadar** [comment]
@stone-w4tch3r thank you so much for working on this! At a quick glance it looks fantastic, time is in extremely short supply right now I won’t be able to properly review till next weekend unfortunately, just a heads up!

File: pyinfra/operations/selinux.py [hunk 1]
```diff
@@ -25,16 +39,19 @@ def boolean(bool_name, value, persistent=False):
         selinux.boolean(
             name='Allow Apache to connect to LDAP server',
             'httpd_can_network_connect',
-            'on',
+            Boolean.ON,
```
Comment (Fizzadar) [inline pyinfra/operations/selinux.py:45]:
```suggestion
            selinux.Boolean.ON,
```

File: pyinfra/operations/selinux.py [hunk 2]
```diff
@@ -127,12 +144,15 @@ def port(protocol, port_num, se_type=None, present=True):
 
         selinux.port(
             name='Allow Apache to provide service on port 2222',
-            'tcp',
+            Protocol.TCP,
```
Comment (Fizzadar) [inline pyinfra/operations/selinux.py:152]:
```suggestion
            selinux.Protocol.TCP,
```

**Review by Fizzadar — COMMENTED** [review]
OK that week turned into a month, my apologies @stone-w4tch3r. PR looks great, HUGE thank you for working on this!

I think the failures are mostly missing `Optional`s since mypy switched to disallowing implicit optional kwargs. Looks like there's [a tool](https://github.com/hauntsaninja/no_implicit_optional) to automatically fix this.

I can clean those up and get that working with the tests, all the types look good 👍

**Comment by Fizzadar** [comment]
Rebased ontop of v3, hoping to get remaining mypy errors cleaned up this week.

**Review by Fizzadar — APPROVED** [review]
Got there in the end! Massive thank you @stone-w4tch3r for doing the majority of the work here, truly a fantastic achievement to land this in v3!

**Pull request merged**